tests/proxy/replicated_map/replicated_map_test.go

```diff
@@ -0,0 +1,323 @@
+// Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package replicated_map
+
+import (
+	"github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/core"
+	. "github.com/hazelcast/hazelcast-go-client/rc"
+	. "github.com/hazelcast/hazelcast-go-client/tests"
+	"log"
+	"reflect"
+	"sort"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+)
+
+var rmp core.ReplicatedMap
+var client hazelcast.IHazelcastInstance
+
+func TestMain(m *testing.M) {
+	remoteController, err := NewRemoteControllerClient("localhost:9701")
+	if remoteController == nil || err != nil {
+		log.Fatal("create remote controller failed:", err)
+	}
+	cluster, err := remoteController.CreateCluster("3.9", DEFAULT_XML_CONFIG)
+	remoteController.StartMember(cluster.ID)
+	client, _ = hazelcast.NewHazelcastClient()
+	rmp, _ = client.GetReplicatedMap("myReplicatedMap")
+	m.Run()
+	rmp.Clear()
+	client.Shutdown()
+	remoteController.ShutdownCluster(cluster.ID)
+}
+
+func TestReplicatedMapProxy_Destroy(t *testing.T) {
+	testKey := "testingKey"
+	testValue := "testingValue"
+	rmp.Put(testKey, testValue)
+	rmp.Destroy()
+	rmp, _ := client.GetReplicatedMap("myReplicatedMap")
+	res, err := rmp.Get(testKey)
+	AssertNilf(t, err, res, "replicatedMap Destroy() failed")
+}
+
+func TestReplicatedMapProxy_Put(t *testing.T) {
```

Comment:
There are missing tests like TestX_withNil tests

Reply:
@asimarslan  I dont think we need those tests here, they go to the same method which is already tested.

Reply:
Those validation methods are tested but your proxy method may not use it correctly so We need to test if a proxy method reject a nil value or not. So we do need these tests.